src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -362,10 +362,22 @@ private function parseDefinition(\DOMElement $service, string $file, array $defa
             $definition->setBindings($bindings);
         }
 
-        if ($value = $service->getAttribute('decorates')) {
+        if ($decorates = $service->getAttribute('decorates')) {
+            $onDecoratedInvalid = $service->getAttribute('decoration-on-invalid') ?: 'exception';
+            if ('exception' === $onDecoratedInvalid) {
+                $invalidBehavior = ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE;
+            } elseif ('ignore' === $onDecoratedInvalid) {
+                $invalidBehavior = ContainerInterface::IGNORE_ON_INVALID_REFERENCE;
+            } elseif ('null' === $onDecoratedInvalid) {
+                $invalidBehavior = ContainerInterface::NULL_ON_INVALID_REFERENCE;
+            } else {
+                throw new InvalidArgumentException(sprintf('Invalid attribute "decoration-on-invalid" defined for "%s" in "%s". Allowed values are "exception", "ignore" and "null".', (string) $service->getAttribute('id'), $file));
```

Comment:
`'Invalid value "%s" for attribute "decoration-on-invalid" on service "%s". Did you mean "exception", "ignore" or "null" in "%s"?'`

(similar in YamlFileLoader)